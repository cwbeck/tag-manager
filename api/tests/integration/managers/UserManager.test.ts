import ManagerTestUtils from '../../utils/ManagerTestUtils';
import TestUtils from '../../utils/TestUtils';

describe('UserManager', () => {
    beforeEach(ManagerTestUtils.beforeEachPrepareContainer);
    afterEach(TestUtils.afterEachRestoreContainerMocksAndDate);

    it('get me', async () => {
        ManagerTestUtils.mockAsLoggedUser();

        const server = TestUtils.createApolloServerForIntegration();

        const Q = `
            query {
                me {
                    id
                    first_name
                    last_name
                    email
                    is_admin
                }
            }
        `;

        // run query against the server and snapshot the output
        const res = await server.executeOperation({ query: Q });
        expect(res).toMatchSnapshot();
    });
});
