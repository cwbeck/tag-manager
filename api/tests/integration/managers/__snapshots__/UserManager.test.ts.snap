// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserManager get me 1`] = `
Object {
  "data": Object {
    "me": Object {
      "email": "test@mail.com",
      "first_name": "FirstName",
      "id": "5edf4c48e9615d9a1c66baaa",
      "is_admin": false,
      "last_name": "LastName",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
